File: Java/0807_Max_Increase_to_Keep_City_Skyline.java
class Solution {
    public int maxIncreaseKeepingSkyline(int[][] grid) {
        int n = grid.length;
        int[] x = new int[n];
        int[] y = new int[n];
        for(int i = 0; i < n; i++){
            for(int j = 0; j < n; j++){
                x[i] = Math.max(x[i], grid[i][j]);
                y[j] = Math.max(y[j], grid[i][j]);
            }
        }
        int result = 0;
        for(int i = 0; i < n; i++){
            for(int j = 0; j < n; j++){
                result += Math.min(x[i], y[j]) - grid[i][j];
            }
        }
        return result;
    }
}

File: CS/0807_Max_Increase_to_Keep_City_Skyline.cs
public class Solution {
    public int MaxIncreaseKeepingSkyline(int[][] grid) {
        int result = 0;
        int n = grid.Length;
        int[] maxX = new int[n];
        int[] maxY = new int[n];
        for(int i = 0; i < n; i++){
            for(int j = 0; j < n; j++){
                maxX[i] = Math.Max(maxX[i], grid[i][j]);
                maxY[j] = Math.Max(maxY[j], grid[i][j]);
            }
        }
        for(int i = 0; i < n; i++){
            for(int j = 0; j < n; j++){
                result += Math.Min(maxX[i], maxY[j]) - grid[i][j];
            }
        }
        return result;
    }
}
